core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -1206,13 +1206,19 @@ open class TileInfo : IsPartOfGameInfoSerialization {
     fun setPillaged() {
         // http://well-of-souls.com/civ/civ5_improvements.html says that naval improvements are destroyed upon pillage
         //    and I can't find any other sources so I'll go with that
-        if (isLand) {
-            // Setting turnsToImprovement might interfere with UniqueType.CreatesOneImprovement
-            removeCreatesOneImprovementMarker()
-            improvementInProgress = improvement
+        if (improvement == null) {
+            improvementInProgress = roadStatus.name
+            roadStatus = RoadStatus.None
             turnsToImprovement = 2
+        } else {
+            if (isLand) {
+                // Setting turnsToImprovement might interfere with UniqueType.CreatesOneImprovement
+                removeCreatesOneImprovementMarker()
+                improvementInProgress = improvement
+                turnsToImprovement = 2
+            }
+            improvement = null
```

Comment:
So if you pillage the same tile twice, the original improvement is fully removed? That differs from civ 5 behaviour iirc

Reply:
Correct. Need to overhaul the system to handle marking improvements (and routes) as pillaged without removing them. Which is something I started but is a pretty huge overhaul